src/localplayer.cpp

```diff
@@ -639,86 +639,86 @@ void LocalPlayer::applyControl(float dtime)
 	if (continuous_forward)
 		speedH += move_direction;
 
-	if (control.up) {
-		if (continuous_forward) {
-			if (fast_move)
-				superspeed = true;
-		} else {
-			speedH += move_direction;
+	if (m_knockback_timer > 0.0f) {
+		m_knockback_timer -= dtime;
+		if (m_knockback_timer <= 0.0f) {
+			m_speed = v3f(0, -9.81 * BS, 0);
```

Comment:
v3f(0, -9.81 * BS, 0); should be the acceleration applied throughout the flight of theplayer, this is acceleration due to gravity not velocity.